### PR TITLE
store: Add metrics for pd api call time

### LINKED ISSUE #29821: Add Metrics to Monitor PD API call exection time
This issue was reported by @AilinKid 

We should support the ability to time PD API calls so that we can better investigate user reported issues with placement rule DDL. There might be some scenarios where PD API call time takes longer than expected.

### PR DESCRIPTION
**Pull request by jyz0309**
Signed-off-by: jyz0309 <45495947@qq.com>

### What problem does this PR solve?

Issue Number: close #29821 

Problem Summary:

### What is changed and how it works?
Add Metrics to Monitor PD API call exection time 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
Add Metrics to Monitor PD API call exection time 
```

### CODE REVIEW

**Comment by morgo** [comment]
LGTM (non binding; I am not much of a prometheus user.)

**Review by xhebox — CHANGES_REQUESTED** [review]
LGTM, but this is not what we want for the placement issue. There are more than one occurences of PD API.

https://github.com/pingcap/tidb/blob/d3a4187f7e99a0c156912b6831686e89fc8b6a85/domain/infosync/info.go#L333-L380

FYI, for placement rules, we primaryly want to monitor this function, instead.

**Review by AilinKid — APPROVED** [review]
LGTM

File: metrics/server.go [hunk 1]
```diff
@@ -229,6 +229,15 @@ var (
...
 		}, []string{LblType, LblResult})
+
+	PDApiExecutionHistogram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "tidb",
+			Subsystem: "server",
+			Name:      "pd_api_execution_duration_seconds",
+			Help:      "Bucketed histogram of all pd api execution time (s)",
+			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20), // 1ms ~ 524s
+		}, []string{LblType})
```
Comment (jyz0309) [inline metrics/server.go:240]: I add a `enum{common, placement}` label for this metric. Plz correct me if you have a better idea. @xhebox

File: store/helper/helper.go [hunk 2]
```diff
@@ -767,10 +768,12 @@ func (h *Helper) requestPD(method, uri string, body io.Reader, res interface{})
 	if err != nil {
 		return err
 	}
+	start := time.Now()
 	resp, err := util.InternalHTTPClient().Do(req)
 	if err != nil {
 		return errors.Trace(err)
 	}
+	metrics.PDApiExecutionHistogram.WithLabelValues("common").Observe(time.Since(start).Seconds())
```
Comment (xhebox) [inline store/helper/helper.go:776]: You did not count the errored requests in. What about moving observe before `err ...` condition branch.

**Review by xhebox — APPROVED** [review]

**Pull request merged**